### PR TITLE
metrics[batch-submitter]: add new batch submitter metrics

### PR DESCRIPTION
**Pull request by annieke**
**Description**
This PR adds additional metrics, mostly failure counts, to the batch submitter metrics to aid our debugging efforts.

**Additional context**
The `failedSubmissions` metric will be updated in the new code path introduced in this PR: https://github.com/ethereum-optimism/optimism/pull/1051 will edit each PR based on whichever is merged first.

**Metadata**
- Fixes OP-869

### CODE REVIEW

**Review by optimisticben — DISMISSED** [review]
looks awesome to me

File: packages/batch-submitter/src/batch-submitter/batch-submitter.ts [hunk 1]
```diff
@@ -274,6 +278,21 @@ export abstract class BatchSubmitter {
         help: 'Gas used to submit each batch',
         registers: [metrics.registry],
       }),
+      batchesSubmitted: new metrics.client.Counter({
+        name: 'batches_submitted',
+        help: 'Count of batches submitted',
+        registers: [metrics.registry],
+      }),
+      failedSubmissions: new metrics.client.Counter({
```
Comment (snario) [inline packages/batch-submitter/src/batch-submitter/batch-submitter.ts:286]: Is this used?

**Review by snario — APPROVED** [review]

File: packages/batch-submitter/src/batch-submitter/batch-submitter.ts (hunk 1 above)
Comment (annieke) [reply packages/batch-submitter/src/batch-submitter/batch-submitter.ts:306]: updated after the error handling was merged!

**Review by optimisticben — APPROVED** [review]
LGTM

**Pull request merged**